llms/local/localllm.go

```diff
@@ -60,15 +60,27 @@ func (o *LLM) Generate(prompts []string) ([]*llms.Generation, error) {
 // New creates a new local LLM implementation.
 func New() (*LLM, error) {
 	// Require the user to set the path to the local LLM binary
-	bin := os.Getenv(localLLMBinVarName)
-	path, err := exec.LookPath(bin)
-	if err != nil {
-		return nil, errors.Join(ErrMissingBin, err)
-	}
+	binPath := os.Getenv(localLLMBinVarName)
 
 	// Allow the user to pass CLI arguments to the local LLM binary (optional)
 	args := os.Getenv(localLLMArgsVarName)
-	c, err := localclient.New(path, args)
+
+	// Expand tilde in the path
+	if binPath[0] == '~' {
+		binPath = "$HOME" + binPath[1:]
```

Comment:
Can you write some tests against it?
And it should be proper to use `filepath.Join(...)` to concat the path elements.

```suggestion
		binPath = filepath.Join("$HOME", strings.TrimPrefix(binPath, "~"))
```

Btw, if you choose to accept my suggestion, you should also import the `filepath` package as well within the `import` section.